GameOfLifeApp/Services/Serializer.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Newtonsoft.Json;
+
+namespace GameOfLifeApp
+{
+    /// <summary>
+    /// Saves JSON data to a file.    
+    /// </summary>
+    public class Serializer : ISerializer
+    {
+        public Serializer()
+        {
+            //data = gameLogic;
+            //pr = properties;
+        }
+
+        // Declaring a private string. Get the complete path of the current working directory. Using CurrentDirectory property of Environment class.
+        private string _fileName = $"{Environment.CurrentDirectory}\\GameData.json";
+
+        /// <summary>
+        /// Creates a StreamWriter and adds some text to the writer using StreamWriter.
+        /// </summary>
+        /// <param name="gameLogic">Saves the data of the object.</param>
+        public void SaveData(IGameField pr)
+        {
+            using (StreamWriter writer = File.CreateText(_fileName))
+            {
+                string output = JsonConvert.SerializeObject(pr);
+                writer.Write(output);
+            }
+        }
+
+        public void Load()
+        {
+            // Open the file for reading.
+            using (StreamReader sr = File.OpenText(_fileName))
+            {
+                //Reads all characters from the current position to the end of the stream and returns them as a single string.
+                string fileText = sr.ReadToEnd();
+                GameField? pr = JsonConvert.DeserializeObject<GameField>(fileText);
```

Comment:
what happens when you load a game, if it deserializes the game but is not passed further?
and why do you use ? is an object by default is null?

Reply:
why the name pr, please update it everywhere to a understandable name.

Reply:
pr is changed into field.

Reply:
Do you mean to continue the game after deserialization?

Reply:
in general if you restore the game it then should continue that game.

Reply:
Done.